libgalois/test/verify-triangle-counting.cpp

```diff
@@ -0,0 +1,43 @@
+#include "katana/PropertyGraph.h"
+#include "katana/SharedMemSys.h"
+#include "katana/analytics/triangle_count/triangle_count.h"
+
+std::unique_ptr<katana::PropertyGraph>
+MakeGridWithDiagonals() noexcept {
+  katana::SymmetricGraphTopologyBuilder builder;
+  builder.AddNodes(4);
+
+  // Build a square
+  builder.AddEdge(0, 1);
+  builder.AddEdge(1, 2);
+  builder.AddEdge(2, 3);
+  builder.AddEdge(3, 0);
+
+  // Add two diagonals, each creating 2 triangles (total 4)
+  builder.AddEdge(0, 2);
+  builder.AddEdge(1, 3);
+
+  katana::GraphTopology topo = builder.ConvertToCSR();
+
+  auto res = katana::PropertyGraph::Make(std::move(topo));
+  KATANA_LOG_ASSERT(res);
+  return std::move(res.value());
+}
+
+int
+main() {
+  katana::SharedMemSys S;
+
+  std::unique_ptr<katana::PropertyGraph> pg = MakeGridWithDiagonals();
+
+  katana::analytics::TriangleCountPlan plan =
+      katana::analytics::TriangleCountPlan::NodeIteration(
```

Comment:
Why only test NodeIteraton algo? Why not OrderedCount and EdgeIteration? BTW OrderedCount is the best and is used by default.

Reply:
done.